use-cases.md

```diff
@@ -44,41 +44,43 @@ These two examples are intentionally naive. Usually developers will not want to
-  1. Finally, close the span using `span.finish()` when the server has finished processing the request.
+  1. Attempt to join an existing trace given a Span that's been propagated alongside the incoming request (in case the trace has already been started by the client), or create a new trace if no such propagated Span could be found.
+  1. Store the newly created Span in some _request context_ that is propagated throughout the application, either by application code, or by the RPC framework.
+  1. Finally, close the Span using `span.finish()` when the server has finished processing the request.
 
-#### Decoding Trace Context from Incoming Request
+#### Joining to a Trace from an Incoming Request
 
-Let's assume that we have an HTTP server, and the Trace Context is propagated from the client via HTTP headers, accessible via `request.headers`:
+Let's assume that we have an HTTP server, and the Span is propagated from the client via HTTP headers, accessible via `request.headers`:
 
 ```python
-    context = tracer.trace_context_from_text(
-        trace_context_id=request.headers,
+    span = tracer.join_trace_from_text(
+        operation_name=operation,
+        context_snapshot=request.headers,
         trace_attributes=request.headers
     )
 ```
 
-Here we set both arguments of the decoding method to the `headers` map. The Tracer object knows which headers it needs to read in order to reconstruct Trace Context, which comprises a trace context ID and trace attributes.
+Here we set both arguments of the decoding method to the `headers` map. The Tracer object knows which headers it needs to read in order to reconstruct the context snapshot and any Trace Attributes.
 
-#### Starting a Server-Side Span
+The `operation` above refers to the name the server wants to give to the Span. For example, if the HTTP request was a POST against `/save_user/123`, the operation name can be set to `post:/save_user/`. OpenTracing API does not dictate how applications name the spans.
 
-The `context` object above can be `None`, if the Tracer did not find relevant headers in the incoming request, either because the client did not send them, or maybe the client is running with a different tracer implementation (e.g. Zipkin vs. AppDash). In this case the server needs to start a brand new trace, otherwise it should join the existing trace:
+#### Joining to or Starting a Trace from an Incoming Request
+
+The `span` object above can be `None` if the Tracer did not find relevant headers in the incoming request: presumably because the client did not send them. In this case the server needs to start a brand new trace.
 
 ```python
-    if context is None:
+    span = tracer.join_trace_from_text(
+        operation_name=operation,
+        context_snapshot=request.headers,
+        trace_attributes=request.headers
+    )
+    if span is None:
```

Comment:
in fact, I remember it came up once with Hyperbahn/TChannel, where Hyperbahn would peek into the trace ID, and if present it would start a new span and replace the trace id in the message being routed (that's the reason why trace ID was stored in a known fixed offset in the message, not in the app headers), but if there is no trace ID or it's not sampled, it would leave it alone.